### ISSUE TITLE
Missing parser error for arguments in class field

### ISSUE THREAD

**Issue by bvanjoi** [microsoft/TypeScript#62062]
### 🔎 Search Terms

parse, class field, arguments

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about _________

### ⏯ Playground Link

https://www.typescriptlang.org/play/?ts=5.8.3#code/EQVwzgpgBGAuBOBLAxrYBuAUJ5B7AdnFAIbwDmIAthPrGFALxQAsATFjgDbFj0CCUAN6YooqADNcuRiXJUadKAHolMABa4QnACZRYa+LgDuUAA6lIUCPEPwANFABGIWFHy5YmAL5A

### 💻 Code

```ts
"use strict";

const arguments = 42;

class A {
    foo = arguments // should throw parse error, but not
}
```

### 🙁 Actual behavior

Syntax error not thrown.

### 🙂 Expected behavior

Throw syntax error.

### Additional information about the issue

<img width="711" height="76" alt="Image" src="https://github.com/user-attachments/assets/2b0177c1-bf18-4995-901b-015df7a81cf7" />

**Comment by MartinJohns** [comment]
IMO that's working as intended. It does show an error when you remove your local `arguments` variable, but with it the parser/compiler assumes you refer to the variable - not the global `arguments`.

**Comment by RyanCavanaugh** [comment]
"early errors" like this are not necessarily reported as a parse error. As long as we issue an error somewhere, that's sufficient
